### PR TITLE
"Registering SwissLipids identifiers in Wikidata"

### PR DESCRIPTION
**Pull request by egonw**
per requested port.

### CODE REVIEW

File: content/recipes/interoperability/registering-swisslipids-identifiers.md [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+:recipe_type: hands_on
+:reading_time_minutes: 15
+:intended_audience: bioinformatician, data_scientist, data_engineer
+:has_executable_code: nope
+:recipe_name: Registering SwissLipids identifiers in Wikidata
+```` 
+
+## Main Objectives
+
+The objective of this recipe is to describe how external identifiers can be added to Wikidata in an automated.
```
Comment (proccaserra) [inline content/recipes/interoperability/registering-swisslipids-identifiers.md]:
```suggestion
The objective of this recipe is to describe how [Wikidata](https://www.wikidata.org) can be augmented by adding `external identifiers`, that is identifiers minted by other authorities for an entity described in Wikidata,  in an automated way. This is to make the content more findable and more interoperable.
```

File: content/recipes/interoperability/registering-swisslipids-identifiers.md [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+
+___
+
+
+## Requirements
+
+To record identifiers of some resource to make the content more findable and more interoperable, first a [Wikidata](https://www.wikidata.org/) property
+needs to be requested. That process is described [here](https://www.wikidata.org/wiki/Wikidata:Property_proposal).
```
Comment (proccaserra) [inline content/recipes/interoperability/registering-swisslipids-identifiers.md]:
```suggestion
- Familiarity with the process of requesting a new property from [Wikidata](https://www.wikidata.org/).
That process is described [here](https://www.wikidata.org/wiki/Wikidata:Property_proposal).
- In order to record identifiers of some resource, first a  property needs to be requested.
```

File: content/recipes/interoperability/registering-swisslipids-identifiers.md [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+
+---
+
+
+## Main Content
+
+Making datasets more findable and databases more interoperable, it is recommended to share not just top-level metadata, but also record-level
```
Comment (proccaserra) [inline content/recipes/interoperability/registering-swisslipids-identifiers.md]: a bit unclear as it leads to the question "what is top-level metadata and what is record-level metadata? ". I would suggest rephrasing or clarifying what you mean here.

File: content/recipes/interoperability/registering-swisslipids-identifiers.md [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+metadata. For example, for chemical databases, you can share your chemical compound identifiers as open data, while keeping the data itself
+in your own database. That allows other databases, like Wikidata, to link to your database. First, that makes your data more findable,
+but it also makes your data easier to integrate and therefore more interoperable.
+
+This recipe follows this idea and starts with a CCZero-licensed mapping file, linking InChIKeys and database identifiers.
```
Comment (proccaserra) [inline content/recipes/interoperability/registering-swisslipids-identifiers.md]:
```suggestion
This recipe follows this idea and starts with a mapping file that links InChIKeys and database identifiers and which is shared under a [CC0 license](https://creativecommons.org/share-your-work/public-domain/cc0/).
```

File: content/recipes/interoperability/registering-swisslipids-identifiers.md [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+Recipe *InChI and SMILES identifiers for chemical structures* ([fcb:FCB007](https://w3id.org/faircookbook/FCB007)) explains how InChIKeys
```
Comment (proccaserra) [inline content/recipes/interoperability/registering-swisslipids-identifiers.md]:
```suggestion
The FAIR cookbook Recipe *[InChI and SMILES identifiers for chemical structures (fcb:FCB007)](https://w3id.org/faircookbook/FCB007)*  explains how InChIKeys
```

File: content/recipes/interoperability/registering-swisslipids-identifiers.md [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+can be generated for chemical compounds in your database.
+
+As an example, this recipe demonstrates the approach of listing [SwissLipid](http://www.swisslipids.org/#/) {footcite}`Aimo2015SwissLipids`
```
Comment (proccaserra) [inline content/recipes/interoperability/registering-swisslipids-identifiers.md]:
```suggestion
To illustrate our point, this recipe demonstrates the approach of listing [SwissLipid](http://www.swisslipids.org/#/) {footcite}`Aimo2015SwissLipids`
```

File: content/recipes/interoperability/registering-swisslipids-identifiers.md [hunk 7]
```diff
@@ -0,0 +1,162 @@
...
+identifiers in Wikidata, developed at the 2021 BioHackathon Europe. Earlier, a [Swiss Lipid property for Wikidata](https://www.wikidata.org/wiki/Wikidata:Property_proposal/SwissLipids_identifier)
```
Comment (proccaserra) [inline content/recipes/interoperability/registering-swisslipids-identifiers.md]:
```suggestion
identifiers in Wikidata, developed at the 2021 BioHackathon Europe.
The work relied on the fact that a [`Swiss Lipids property` for Wikidata](https://www.wikidata.org/wiki/Wikidata:Property_proposal/SwissLipids_identifier)
```

File: content/recipes/interoperability/registering-swisslipids-identifiers.md [hunk 8]
```diff
@@ -0,0 +1,162 @@
...
+was already proposed, later approved, and created in Wikidata [just before the 2020 BioHackathon Europe](https://www.wikidata.org/w/index.php?title=Property:P8691&oldid=1287579005).
+An existing Wikidata property is a requirement to adding the external identifiers.
```
Comment (proccaserra) [inline content/recipes/interoperability/registering-swisslipids-identifiers.md:47]:
````suggestion

```{warning}
An existing Wikidata property is a requirement to adding the external identifiers.
```

```

File: content/recipes/interoperability/registering-swisslipids-identifiers.md [hunk 9]
```diff
@@ -0,0 +1,162 @@
...
+
+The recipe uses [Bacting](https://github.com/egonw/bacting) {footcite}`Willighagen2021` which can be used in Groovy and Python. We
```
Comment (proccaserra) [inline content/recipes/interoperability/registering-swisslipids-identifiers.md]: @egon: Bacting isn't available from biotools. Would you be able to submit and link back to this recipe?

File: content/recipes/interoperability/registering-swisslipids-identifiers.md [hunk 10]
```diff
@@ -0,0 +1,162 @@
...
+will use the first here. The starting point is mappings (tuples) that link the SwissLipid identifiers to InChIKeys. The latter
```
Comment (proccaserra) [inline content/recipes/interoperability/registering-swisslipids-identifiers.md]:
```suggestion
will use the first here.

The starting point is mappings (tuples) that link the `SwissLipids` identifiers to `InChIKeys`.
The latter
```

File: content/recipes/interoperability/registering-swisslipids-identifiers.md [hunk 11]
```diff
@@ -0,0 +1,162 @@
...
+is used to find the matching Wikidata items.
+
+### Step 1: getting the data
+
+The data is CC-BY but it was agreed that adding the SwissLipid identifiers to Wikidata (CCZero) is okay.
```
Comment (proccaserra) [inline content/recipes/interoperability/registering-swisslipids-identifiers.md]:
```suggestion
While the SwissLipids data is available under [CC-BY license](https://creativecommons.org/licenses/by/4.0/), it was agreed that adding the SwissLipids identifiers to Wikidata resource, the data of which is available under [CC0 license](https://creativecommons.org/publicdomain/zero/1.0/) is legitimate.
```

File: content/recipes/interoperability/registering-swisslipids-identifiers.md [hunk 12]
```diff
@@ -0,0 +1,162 @@
...
+
+* Download `lipids.tsv` from the [Downloads](https://www.swisslipids.org/#/downloads) page
+* Gunzip the file
+
+### Step 2: extract Swiss Lipid ID <> InChIKey tuples
+
+For this step I use `csvtool` (`apt get install csvtool`):
```
Comment (proccaserra) [inline content/recipes/interoperability/registering-swisslipids-identifiers.md]:
```suggestion
For this step, I use `csvtool` (`apt get install csvtool`):
```

File: content/recipes/interoperability/registering-swisslipids-identifiers.md [hunk 13]
```diff
@@ -0,0 +1,162 @@
...
+
+This results in almost 600k tuples:
+
+```shell
+$ wc -l swiss*tsv
+   592412 swisslipids_ids.tsv
+   777957 swisslipids.tsv
+```
+
+### Step 3: creating a ShEx model
```
Comment (proccaserra) [inline content/recipes/interoperability/registering-swisslipids-identifiers.md]: @egonw:
may be add a statement regarding the selection of shex over a W3C standard (shacl) (even if converters exist).

File: content/recipes/interoperability/registering-swisslipids-identifiers.md [hunk 14]
```diff
@@ -0,0 +1,162 @@
...
+
+Skipping this step for later this week, but here the task is to create a shape expression for Wikidata, to model how
+the identifiers will be added to Wikidata. See the recept publication by Waagmeester et al. about _A protocol for adding knowledge to Wikidata: aligning resources on human coronaviruses_
+{footcite}`Waagmeester2021`.
+
+### Step 4: creating QuickStatements
+
+Now we have the mappings and the data model in Wikidata, we can create QuickStatements to allow us to enter the
```
Comment (proccaserra) [inline content/recipes/interoperability/registering-swisslipids-identifiers.md]:
```suggestion
Now we have the mappings and the data model in Wikidata, we can create [QuickStatements](https://www.wikidata.org/wiki/Help:QuickStatements) to allow us to enter the
```

File: content/recipes/interoperability/registering-swisslipids-identifiers.md [hunk 15]
```diff
@@ -0,0 +1,162 @@
...
+data into Wikidata. This is not the only approach, and the process can be further automated using "Wikidata bots".
+For this, see BioHackathon Europe [Project 32](https://github.com/elixir-europe/biohackathon-projects-2021/tree/main/projects/32):
+Connecting ELIXIR-related open data on Wikidata via [WikiProject ELIXIR](https://www.wikidata.org/wiki/Wikidata:WikiProject_Elixir).
+
+Based on existing Bacting scripts, a script is created to take the `swisslipids_ids.tsv` file as input and create
+QuickStatements: [https://github.com/egonw/ons-wikidata/blob/master/ExtIdentifiers/swisslipids.groovy](https://github.com/egonw/ons-wikidata/blob/master/ExtIdentifiers/swisslipids.groovy) This script is using [Apache Groovy](http://www.groovy-lang.org/)
+but Bacting can also be using in Python, see [https://github.com/cthoyt/pybacting](https://github.com/cthoyt/pybacting).
+
+This script uses a federated query against https://beta.sparql.swisslipids.org/sparql/ after a suggestion by Dr [Jerven Bolleman](https://orcid.org/0000-0002-7449-1266)
```
Comment (proccaserra) [inline content/recipes/interoperability/registering-swisslipids-identifiers.md:108]: @egonw -> add Jergen to the list of authors/contributors.

File: content/recipes/interoperability/registering-swisslipids-identifiers.md [hunk 16]
```diff
@@ -0,0 +1,162 @@
...
+### Step 4: creating QuickStatements
+
+Now we have the mappings and the data model in Wikidata, we can create QuickStatements to allow us to enter the
+data into Wikidata. This is not the only approach, and the process can be further automated using "Wikidata bots".
+For this, see BioHackathon Europe [Project 32](https://github.com/elixir-europe/biohackathon-projects-2021/tree/main/projects/32):
+Connecting ELIXIR-related open data on Wikidata via [WikiProject ELIXIR](https://www.wikidata.org/wiki/Wikidata:WikiProject_Elixir).
+
+Based on existing Bacting scripts, a script is created to take the `swisslipids_ids.tsv` file as input and create
+QuickStatements: [https://github.com/egonw/ons-wikidata/blob/master/ExtIdentifiers/swisslipids.groovy](https://github.com/egonw/ons-wikidata/blob/master/ExtIdentifiers/swisslipids.groovy) This script is using [Apache Groovy](http://www.groovy-lang.org/)
+but Bacting can also be using in Python, see [https://github.com/cthoyt/pybacting](https://github.com/cthoyt/pybacting).
```
Comment (proccaserra) [inline content/recipes/interoperability/registering-swisslipids-identifiers.md:106]: @egonw : what about adding a jupyter notebook using pybacting?

File: content/recipes/interoperability/registering-swisslipids-identifiers.md [hunk 17]
```diff
@@ -0,0 +1,162 @@
...
+### Step 5: running the QuickStatements
+
+The resulting mappings are being added via the [QuickStatements website](https://quickstatements.toolforge.org/).
+
+## Conclusion
+
+It will be clear Bacting can be replaced without too much effort by other tools making the same steps. What is the
+main conclusion is that we can indeed automate adding external identifiers routinely to Wikidata using InChIKey
+matching.
+
```
Comment (proccaserra) [inline content/recipes/interoperability/registering-swisslipids-identifiers.md]:
```suggestion
The main conclusion is that it is possible to automate adding external / third party identifiers routinely to Wikidata using InChIKey matching.
The tool `Bacting` used in the example can obviously be replaced without too much effort by any other tools performing the same steps.

```

File: content/recipes/interoperability/registering-swisslipids-identifiers.md (hunk 16 above)
Comment (egonw) [reply content/recipes/interoperability/registering-swisslipids-identifiers.md:106]: yes, I like that idea. I had to get more used to this but found my way around now, including booting a notebook in Google collab: https://bioclipse.github.io/bioclipse.scripting/code/ParseIUPACName.code.html

Comment (egonw) [reply content/recipes/interoperability/registering-swisslipids-identifiers.md:106]: I will do that in an update. I was thinking of doing this for https://faircookbook.elixir-europe.org/content/recipes/infrastructure/chemical-identities.html too

File: content/recipes/interoperability/registering-swisslipids-identifiers.md (hunk 9 above)
Comment (egonw) [reply content/recipes/interoperability/registering-swisslipids-identifiers.md]: Good question. I need to check. In the past, bio.tools was pretty much about services, not software. Did this change?

File: content/recipes/interoperability/registering-swisslipids-identifiers.md [hunk 18]
```diff
@@ -0,0 +1,171 @@
+# Registering SwissLipids identifiers in Wikidata
+
+ ````{panels_fairplus}
+:identifier_text: FCBxxx
+:identifier_link: 'https://w3id.org/faircookbook/FCBxxx'
+:difficulty_level: 5
+:recipe_type: hands_on
+:reading_time_minutes: 15
+:intended_audience: bioinformatician, data_scientist, data_engineer
+:has_executable_code: nope
```
Comment (egonw) [inline content/recipes/interoperability/registering-swisslipids-identifiers.md:10]: fixme

File: content/recipes/interoperability/registering-swisslipids-identifiers.md (hunk 9 above)
Comment (egonw) [reply content/recipes/interoperability/registering-swisslipids-identifiers.md]: Okay, I misremembered that. They indeed support just software fully too. I'll make an entry!

Comment (egonw) [reply content/recipes/interoperability/registering-swisslipids-identifiers.md]: https://bio.tools/bacting

File: content/recipes/interoperability/registering-swisslipids-identifiers.md (hunk 18 above)
Comment (egonw) [reply content/recipes/interoperability/registering-swisslipids-identifiers.md:10]: lol: https://github.com/FAIRplus/the-fair-cookbook/blob/cdaa3931a60c573b971b5b8be1820820974a78c2/_ext/panels_fairplus.py#L125

File: content/recipes/interoperability/registering-swisslipids-identifiers.md (hunk 13 above)
Comment (egonw) [reply content/recipes/interoperability/registering-swisslipids-identifiers.md:87]: Sorry I missed this, but this step is actually still empty. I will remove it for now.

Comment (egonw) [reply content/recipes/interoperability/registering-swisslipids-identifiers.md]: Added SHACL and updated the text a bit.

File: content/recipes/interoperability/registering-swisslipids-identifiers.md (hunk 15 above)
Comment (egonw) [reply content/recipes/interoperability/registering-swisslipids-identifiers.md:108]: It doesn't feel right to have him under the section "Authors". I will add an Acks section, as I want to ack the ELIXIR BioHackathon and SwissLipid too.

**Comment by egonw** [comment]
Moved to https://github.com/FAIRplus/the-fair-cookbook/pull/424

**Pull request closed without merging**